File: SimpleXPCService/XPCService/XPCService.m

//
//  XPCService.m
//  XPCService
//
//  Created by Admin on 13.09.15.
//  Copyright (c) 2015 Admin. All rights reserved.
//

#import "XPCService.h"

@implementation SimpleService //: NSObject <SimpleProtocol>

// This implements the example protocol. Replace the body of this class with the implementation of this service's protocol.
- (void)upperCaseString:(NSString *)aString withReply:(void (^)(NSString *))reply {
    NSString *response = [aString uppercaseString];
    reply(response);
}

- (void)generateSomeNumber:(UInt32)maxNumber withReply:(void (^)(NSString *))reply
{
    int number = arc4random_uniform(maxNumber);
    NSString *result = [[NSString alloc] initWithFormat:@"%d", number];
    reply(result);
}

@end

File: SimpleXPCService/XPCService/XPCService.swift
//
//  XPCService.swift
//  SimpleXPCService
//
//  Created by Admin on 15.09.15.
//  Copyright (c) 2015 Admin. All rights reserved.
//

import Foundation

@objc
class SimpleService: NSObject, SimpleProtocol {
    func generateSomeNumber(maxNumber: UInt32) -> Int {
        let number = Int(arc4random_uniform(maxNumber))
        return number
    }
}